% Options for packages loaded elsewhere
\PassOptionsToPackage{unicode}{hyperref}
\PassOptionsToPackage{hyphens}{url}
%
\documentclass[
]{article}
\usepackage{amsmath,amssymb}
\usepackage{iftex}
\ifPDFTeX
  \usepackage[T1]{fontenc}
  \usepackage[utf8]{inputenc}
  \usepackage{textcomp} % provide euro and other symbols
\else % if luatex or xetex
  \usepackage{unicode-math} % this also loads fontspec
  \defaultfontfeatures{Scale=MatchLowercase}
  \defaultfontfeatures[\rmfamily]{Ligatures=TeX,Scale=1}
\fi
\usepackage{lmodern}
\ifPDFTeX\else
  % xetex/luatex font selection
\fi
% Use upquote if available, for straight quotes in verbatim environments
\IfFileExists{upquote.sty}{\usepackage{upquote}}{}
\IfFileExists{microtype.sty}{% use microtype if available
  \usepackage[]{microtype}
  \UseMicrotypeSet[protrusion]{basicmath} % disable protrusion for tt fonts
}{}
\makeatletter
\@ifundefined{KOMAClassName}{% if non-KOMA class
  \IfFileExists{parskip.sty}{%
    \usepackage{parskip}
  }{% else
    \setlength{\parindent}{0pt}
    \setlength{\parskip}{6pt plus 2pt minus 1pt}}
}{% if KOMA class
  \KOMAoptions{parskip=half}}
\makeatother
\usepackage{xcolor}
\usepackage[margin=1in]{geometry}
\usepackage{color}
\usepackage{fancyvrb}
\newcommand{\VerbBar}{|}
\newcommand{\VERB}{\Verb[commandchars=\\\{\}]}
\DefineVerbatimEnvironment{Highlighting}{Verbatim}{commandchars=\\\{\}}
% Add ',fontsize=\small' for more characters per line
\usepackage{framed}
\definecolor{shadecolor}{RGB}{248,248,248}
\newenvironment{Shaded}{\begin{snugshade}}{\end{snugshade}}
\newcommand{\AlertTok}[1]{\textcolor[rgb]{0.94,0.16,0.16}{#1}}
\newcommand{\AnnotationTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\AttributeTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{#1}}
\newcommand{\BaseNTok}[1]{\textcolor[rgb]{0.00,0.00,0.81}{#1}}
\newcommand{\BuiltInTok}[1]{#1}
\newcommand{\CharTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\CommentTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textit{#1}}}
\newcommand{\CommentVarTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\ConstantTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{#1}}
\newcommand{\ControlFlowTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{\textbf{#1}}}
\newcommand{\DataTypeTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{#1}}
\newcommand{\DecValTok}[1]{\textcolor[rgb]{0.00,0.00,0.81}{#1}}
\newcommand{\DocumentationTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\ErrorTok}[1]{\textcolor[rgb]{0.64,0.00,0.00}{\textbf{#1}}}
\newcommand{\ExtensionTok}[1]{#1}
\newcommand{\FloatTok}[1]{\textcolor[rgb]{0.00,0.00,0.81}{#1}}
\newcommand{\FunctionTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{\textbf{#1}}}
\newcommand{\ImportTok}[1]{#1}
\newcommand{\InformationTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\KeywordTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{\textbf{#1}}}
\newcommand{\NormalTok}[1]{#1}
\newcommand{\OperatorTok}[1]{\textcolor[rgb]{0.81,0.36,0.00}{\textbf{#1}}}
\newcommand{\OtherTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{#1}}
\newcommand{\PreprocessorTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textit{#1}}}
\newcommand{\RegionMarkerTok}[1]{#1}
\newcommand{\SpecialCharTok}[1]{\textcolor[rgb]{0.81,0.36,0.00}{\textbf{#1}}}
\newcommand{\SpecialStringTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\StringTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\VariableTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\VerbatimStringTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\WarningTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\usepackage{graphicx}
\makeatletter
\def\maxwidth{\ifdim\Gin@nat@width>\linewidth\linewidth\else\Gin@nat@width\fi}
\def\maxheight{\ifdim\Gin@nat@height>\textheight\textheight\else\Gin@nat@height\fi}
\makeatother
% Scale images if necessary, so that they will not overflow the page
% margins by default, and it is still possible to overwrite the defaults
% using explicit options in \includegraphics[width, height, ...]{}
\setkeys{Gin}{width=\maxwidth,height=\maxheight,keepaspectratio}
% Set default figure placement to htbp
\makeatletter
\def\fps@figure{htbp}
\makeatother
\setlength{\emergencystretch}{3em} % prevent overfull lines
\providecommand{\tightlist}{%
  \setlength{\itemsep}{0pt}\setlength{\parskip}{0pt}}
\setcounter{secnumdepth}{-\maxdimen} % remove section numbering
\ifLuaTeX
  \usepackage{selnolig}  % disable illegal ligatures
\fi
\IfFileExists{bookmark.sty}{\usepackage{bookmark}}{\usepackage{hyperref}}
\IfFileExists{xurl.sty}{\usepackage{xurl}}{} % add URL line breaks if available
\urlstyle{same}
\hypersetup{
  pdftitle={Microbiome DADA2},
  pdfauthor={Tayler},
  hidelinks,
  pdfcreator={LaTeX via pandoc}}

\title{Microbiome DADA2}
\author{Tayler}
\date{2024-04-18}

\begin{document}
\maketitle

\hypertarget{load-required-packages}{%
\section{Load required packages}\label{load-required-packages}}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{library}\NormalTok{(dada2)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Loading required package: Rcpp
\end{verbatim}

\begin{verbatim}
## Warning: package 'Rcpp' was built under R version 4.2.3
\end{verbatim}

\hypertarget{load-sequences}{%
\section{Load sequences}\label{load-sequences}}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{path }\OtherTok{\textless{}{-}} \StringTok{"sequences"}
\FunctionTok{list.files}\NormalTok{(path)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
##  [1] "filtered"                       "L1S105_9_L001_R1_001.fastq.gz" 
##  [3] "L1S140_6_L001_R1_001.fastq.gz"  "L1S208_10_L001_R1_001.fastq.gz"
##  [5] "L1S257_11_L001_R1_001.fastq.gz" "L1S281_5_L001_R1_001.fastq.gz" 
##  [7] "L1S57_13_L001_R1_001.fastq.gz"  "L1S76_12_L001_R1_001.fastq.gz" 
##  [9] "L1S8_8_L001_R1_001.fastq.gz"    "L2S155_25_L001_R1_001.fastq.gz"
## [11] "L2S175_27_L001_R1_001.fastq.gz" "L2S204_1_L001_R1_001.fastq.gz" 
## [13] "L2S222_23_L001_R1_001.fastq.gz" "L2S240_7_L001_R1_001.fastq.gz" 
## [15] "L2S309_33_L001_R1_001.fastq.gz" "L2S357_15_L001_R1_001.fastq.gz"
## [17] "L2S382_34_L001_R1_001.fastq.gz" "L3S242_19_L001_R1_001.fastq.gz"
## [19] "L3S294_16_L001_R1_001.fastq.gz" "L3S313_32_L001_R1_001.fastq.gz"
## [21] "L3S341_18_L001_R1_001.fastq.gz" "L3S360_4_L001_R1_001.fastq.gz" 
## [23] "L3S378_24_L001_R1_001.fastq.gz" "L4S112_26_L001_R1_001.fastq.gz"
## [25] "L4S137_21_L001_R1_001.fastq.gz" "L4S63_31_L001_R1_001.fastq.gz" 
## [27] "L5S104_28_L001_R1_001.fastq.gz" "L5S155_2_L001_R1_001.fastq.gz" 
## [29] "L5S174_29_L001_R1_001.fastq.gz" "L5S203_3_L001_R1_001.fastq.gz" 
## [31] "L5S222_17_L001_R1_001.fastq.gz" "L5S240_14_L001_R1_001.fastq.gz"
## [33] "L6S20_20_L001_R1_001.fastq.gz"  "L6S68_30_L001_R1_001.fastq.gz" 
## [35] "L6S93_22_L001_R1_001.fastq.gz"  "MANIFEST"                      
## [37] "metadata.yml"
\end{verbatim}

\hypertarget{read-in-file-names}{%
\section{Read in file names}\label{read-in-file-names}}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\# Fastq filenames have format: SAMPLENAME\_R1\_001.fastq. We only have forward reads.}
\NormalTok{fnFs }\OtherTok{\textless{}{-}} \FunctionTok{sort}\NormalTok{(}\FunctionTok{list.files}\NormalTok{(path, }\AttributeTok{pattern=}\StringTok{"\_R1\_001.fastq"}\NormalTok{, }\AttributeTok{full.names =} \ConstantTok{TRUE}\NormalTok{))}

\CommentTok{\# Extract sample names, assuming filenames have format: SAMPLENAME\_XXX.fastq}
\NormalTok{sample.names }\OtherTok{\textless{}{-}} \FunctionTok{sapply}\NormalTok{(}\FunctionTok{strsplit}\NormalTok{(}\FunctionTok{basename}\NormalTok{(fnFs), }\StringTok{"\_"}\NormalTok{), }\StringTok{\textasciigrave{}}\AttributeTok{[}\StringTok{\textasciigrave{}}\NormalTok{, }\DecValTok{1}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\hypertarget{inspect-read-quality}{%
\section{Inspect read quality}\label{inspect-read-quality}}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\# take first 2 forward reads and pop out a graph of read quality}
\FunctionTok{plotQualityProfile}\NormalTok{(fnFs[}\DecValTok{1}\SpecialCharTok{:}\DecValTok{2}\NormalTok{])}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Warning: The `<scale>` argument of `guides()` cannot be `FALSE`. Use "none" instead as
## of ggplot2 3.3.4.
## i The deprecated feature was likely used in the dada2 package.
##   Please report the issue at <]8;;https://github.com/benjjneb/dada2/issueshttps://github.com/benjjneb/dada2/issues]8;;>.
## This warning is displayed once every 8 hours.
## Call `lifecycle::last_lifecycle_warnings()` to see where this warning was
## generated.
\end{verbatim}

\includegraphics{dada2code_files/figure-latex/unnamed-chunk-4-1.pdf}

\hypertarget{filter-and-trim}{%
\section{Filter and trim}\label{filter-and-trim}}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\# Place filtered files in filtered/ subdirectory}
\NormalTok{filtFs }\OtherTok{\textless{}{-}} \FunctionTok{file.path}\NormalTok{(path, }\StringTok{"filtered"}\NormalTok{, }\FunctionTok{paste0}\NormalTok{(sample.names, }\StringTok{"\_F\_filt.fastq.gz"}\NormalTok{))}

\FunctionTok{names}\NormalTok{(filtFs) }\OtherTok{\textless{}{-}}\NormalTok{ sample.names}

\CommentTok{\# quality decreases at 120}
\NormalTok{out }\OtherTok{\textless{}{-}} \FunctionTok{filterAndTrim}\NormalTok{(fnFs, filtFs, }\AttributeTok{truncLen=}\FunctionTok{c}\NormalTok{(}\DecValTok{120}\NormalTok{),}
              \AttributeTok{maxN=}\DecValTok{0}\NormalTok{, }\AttributeTok{maxEE=}\FunctionTok{c}\NormalTok{(}\DecValTok{2}\NormalTok{), }\AttributeTok{truncQ=}\DecValTok{2}\NormalTok{, }\AttributeTok{rm.phix=}\ConstantTok{TRUE}\NormalTok{,}
              \AttributeTok{compress=}\ConstantTok{TRUE}\NormalTok{, }\AttributeTok{multithread=}\ConstantTok{FALSE}\NormalTok{) }\CommentTok{\# On Windows set multithread=FALSE}
\FunctionTok{head}\NormalTok{(out)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
##                                reads.in reads.out
## L1S105_9_L001_R1_001.fastq.gz     11340      8571
## L1S140_6_L001_R1_001.fastq.gz      9738      7677
## L1S208_10_L001_R1_001.fastq.gz    11337      9261
## L1S257_11_L001_R1_001.fastq.gz     8216      6705
## L1S281_5_L001_R1_001.fastq.gz      8907      7067
## L1S57_13_L001_R1_001.fastq.gz     11752      9299
\end{verbatim}

\hypertarget{learn-error-rates-using-machine-learningai}{%
\section{Learn error rates using machine
learning/AI}\label{learn-error-rates-using-machine-learningai}}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{errF }\OtherTok{\textless{}{-}} \FunctionTok{learnErrors}\NormalTok{(filtFs, }\AttributeTok{multithread=}\ConstantTok{FALSE}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## 19539480 total bases in 162829 reads from 34 samples will be used for learning the error rates.
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{plotErrors}\NormalTok{(errF, }\AttributeTok{nominalQ=}\ConstantTok{TRUE}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Warning in scale_y_log10(): log-10 transformation introduced infinite values.
## log-10 transformation introduced infinite values.
\end{verbatim}

\includegraphics{dada2code_files/figure-latex/unnamed-chunk-6-1.pdf} \#
Sample inference (identifying number of unique sequences)

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{dadaFs }\OtherTok{\textless{}{-}} \FunctionTok{dada}\NormalTok{(filtFs, }\AttributeTok{err=}\NormalTok{errF, }\AttributeTok{multithread=}\ConstantTok{FALSE}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Sample 1 - 8571 reads in 2110 unique sequences.
## Sample 2 - 7677 reads in 1728 unique sequences.
## Sample 3 - 9261 reads in 2490 unique sequences.
## Sample 4 - 6705 reads in 1940 unique sequences.
## Sample 5 - 7067 reads in 2144 unique sequences.
## Sample 6 - 9299 reads in 2317 unique sequences.
## Sample 7 - 8395 reads in 1967 unique sequences.
## Sample 8 - 7663 reads in 1573 unique sequences.
## Sample 9 - 4112 reads in 1272 unique sequences.
## Sample 10 - 4546 reads in 1325 unique sequences.
## Sample 11 - 3379 reads in 1131 unique sequences.
## Sample 12 - 3485 reads in 1574 unique sequences.
## Sample 13 - 5183 reads in 1104 unique sequences.
## Sample 14 - 1550 reads in 641 unique sequences.
## Sample 15 - 2526 reads in 874 unique sequences.
## Sample 16 - 4279 reads in 1281 unique sequences.
## Sample 17 - 970 reads in 246 unique sequences.
## Sample 18 - 1313 reads in 483 unique sequences.
## Sample 19 - 1191 reads in 460 unique sequences.
## Sample 20 - 1109 reads in 478 unique sequences.
## Sample 21 - 1132 reads in 603 unique sequences.
## Sample 22 - 1358 reads in 379 unique sequences.
## Sample 23 - 8603 reads in 2252 unique sequences.
## Sample 24 - 10064 reads in 2146 unique sequences.
## Sample 25 - 10096 reads in 2882 unique sequences.
## Sample 26 - 2253 reads in 448 unique sequences.
## Sample 27 - 1828 reads in 379 unique sequences.
## Sample 28 - 1969 reads in 407 unique sequences.
## Sample 29 - 2133 reads in 459 unique sequences.
## Sample 30 - 2556 reads in 468 unique sequences.
## Sample 31 - 1817 reads in 380 unique sequences.
## Sample 32 - 7087 reads in 983 unique sequences.
## Sample 33 - 6169 reads in 1033 unique sequences.
## Sample 34 - 7483 reads in 1272 unique sequences.
\end{verbatim}

\hypertarget{create-sequence-table.-our-output-will-print-the-dimensions-of-our-table}{%
\section{create sequence table. Our output will print the dimensions of
our
table}\label{create-sequence-table.-our-output-will-print-the-dimensions-of-our-table}}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{seqtab }\OtherTok{\textless{}{-}} \FunctionTok{makeSequenceTable}\NormalTok{(dadaFs)}
\FunctionTok{dim}\NormalTok{(seqtab)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1]  34 819
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\# Inspect distribution of sequence lengths}
\FunctionTok{table}\NormalTok{(}\FunctionTok{nchar}\NormalTok{(}\FunctionTok{getSequences}\NormalTok{(seqtab)))}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## 
## 120 
## 819
\end{verbatim}

\hypertarget{remove-chimeras}{%
\section{remove chimeras}\label{remove-chimeras}}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{seqtab.nochim }\OtherTok{\textless{}{-}} \FunctionTok{removeBimeraDenovo}\NormalTok{(seqtab, }\AttributeTok{method=}\StringTok{"consensus"}\NormalTok{, }\AttributeTok{multithread=}\ConstantTok{TRUE}\NormalTok{, }\AttributeTok{verbose=}\ConstantTok{TRUE}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Identified 48 bimeras out of 819 input sequences.
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{dim}\NormalTok{(seqtab.nochim)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1]  34 771
\end{verbatim}

\hypertarget{track-reads-through-pipeline}{%
\section{track reads through
pipeline}\label{track-reads-through-pipeline}}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{getN }\OtherTok{\textless{}{-}} \ControlFlowTok{function}\NormalTok{(x) }\FunctionTok{sum}\NormalTok{(}\FunctionTok{getUniques}\NormalTok{(x))}
\NormalTok{track }\OtherTok{\textless{}{-}} \FunctionTok{cbind}\NormalTok{(out, }\FunctionTok{sapply}\NormalTok{(dadaFs, getN), }\FunctionTok{rowSums}\NormalTok{(seqtab.nochim))}
\FunctionTok{colnames}\NormalTok{(track) }\OtherTok{\textless{}{-}} \FunctionTok{c}\NormalTok{(}\StringTok{"input"}\NormalTok{, }\StringTok{"filtered"}\NormalTok{, }\StringTok{"denoisedF"}\NormalTok{, }\StringTok{"nonchim"}\NormalTok{)}
\FunctionTok{rownames}\NormalTok{(track) }\OtherTok{\textless{}{-}}\NormalTok{ sample.names}
\FunctionTok{head}\NormalTok{(track)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
##        input filtered denoisedF nonchim
## L1S105 11340     8571      8499    7780
## L1S140  9738     7677      7605    7163
## L1S208 11337     9261      9152    8152
## L1S257  8216     6705      6627    6388
## L1S281  8907     7067      6976    6615
## L1S57  11752     9299      9260    8702
\end{verbatim}

\hypertarget{save-seqtab.nochim-as-an-r-file}{%
\section{Save seqtab.nochim as an R
file}\label{save-seqtab.nochim-as-an-r-file}}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{save}\NormalTok{(seqtab.nochim, }\AttributeTok{file=}\StringTok{"RData/seqtab.nochim.RData"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\hypertarget{load-seqtab.nochim-to-start-here}{%
\section{Load seqtab.nochim to start
here}\label{load-seqtab.nochim-to-start-here}}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{load}\NormalTok{(}\StringTok{"RData/seqtab.nochim.RData"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\hypertarget{assign-taxonomy}{%
\section{Assign Taxonomy}\label{assign-taxonomy}}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\# download the Silva species database from https://zenodo.org/records/4587955}
\CommentTok{\# assign taxonomy to a variable called taxa}
\NormalTok{taxa }\OtherTok{\textless{}{-}} \FunctionTok{assignTaxonomy}\NormalTok{(seqtab.nochim, }\StringTok{"silva\_nr99\_v138.1\_wSpecies\_train\_set.fa.gz"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\hypertarget{save-taxonomy-as-a-file}{%
\section{save taxonomy as a file}\label{save-taxonomy-as-a-file}}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{save}\NormalTok{(taxa, }\AttributeTok{file =} \StringTok{"RData/taxa.RData"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}


\end{document}
