---
title: "Phyloseq"
author: "Tayler"
date: "2024-04-18"
output: pdf_document
---

# Install phyloseq
```{r}
if (!require("BiocManager", quietly = TRUE))
    install.packages("BiocManager")

BiocManager::install("phyloseq")
install(BiMiCo)
```

# install peterolah001/BiMiCo: Biomap Microbiome Core methods
```{r}
install.packages("remotes")
install.packages("ggplot2")
```

# Load required packages
```{r}
library(phyloseq)
library(dplyr)
library(ggplot2)
```

```{r}
load("RData/taxa.RData")
load ("RData/seqtab.nochim.RData")
```

# import metadata
```{r}
metadata<- read.csv("metadata.csv", header = TRUE, row.names = 1)
```

# create phyloseq object
```{r}
#make sure the seqtab.nochim and taxa objects are loaded
#create a function called phyloseq. otu_table is from seq table nochim. Sample data found in metadata file. It makes a taxonomy table, too. combine seq data metadata, and +++ into one file called physeq 
physeq <- phyloseq(otu_table(seqtab.nochim, taxa_are_rows=FALSE),
                   sample_data(metadata),
                   tax_table(taxa))
physeq
```
# now we have the raw no. of reads associated w each spp. Like we have 1,000 reads for staph. and 50 enterococc

# transform sample counts. turn raw numbers into percentages so we can compare
```{r}
# convert from raw to abundance
physeq <- transform_sample_counts(physeq, function(abund) 1*(abund>0))
physeq
```
# right now our file has the raw sequencing read (ATCG).

# Remove the sequence itself and replace with ASV. This will pull out the actual sequences into a different line
# this will make it so that the computer does not refer to each sequence in terms of ATCG, it will refer to each seq as "sequence 1" or whatever.
```{r}
dna <- Biostrings::DNAStringSet(taxa_names(physeq))
names(dna) <- taxa_names(physeq)
physeq <- merge_phyloseq(physeq, dna)
taxa_names(physeq) <- paste0("ASV", seq(ntaxa(physeq)))
physeq
```

# remove mitochondria and chloroplast matches. remove all nonbacterial sequences. Remember we took samples from 2 different at different parts of their bodies at different times over time.
# takes object physeq. looks in it to find any sample that has the family as mitochondria or the order as chloroplast. Subset taxa is pulling anything out that matches Mitochondria or chloroplast and its leaving anything else
```{r}
physeq <- physeq %>% subset_taxa( Family!= "Mitochondria" | is.na(Family) &
Order!="Chloroplast" | is.na(Order) )
physeq
```
# we used to have 771 taxa. Now we have 742. ~31 were mt or chlororplast sequences

# remove all non bacterial sequences (archaea)
```{r}
physeq<-rm_nonbac(physeq)
physeq
```

# save physeq objects to load later
```{r}
save(physeq, file="RData/physeq.RData")
```

# load physeq objects to start here
```{r}
load("RData/physeq.RData")
```

# plot bar graph based on phylum
# fill bars based on phylum
# package from gg plot 2 adds an aesthetic where every phylum is a diff color.. stat=identity uses the raw numbers, NOT the percentages.
```{r}
plot_bar(physeq, fill = "Phylum") + geom_bar(aes(color=Phylum, fill=Phylum), stat="identity", position="fill")
```

#create a barplot of relative abundance
```{r}
# convert to relative abundance
physeq_relabund <- transform_sample_counts(physeq, function(x) x / sum(x))

# barplot
plot_bar(physeq_relabund, fill = "Phylum") + geom_bar(aes(color=Phylum, fill=Phylum), stat = "identity", position="fill") + facet_wrap(~body.site, scales = "free")
```

#plot alpha diversity based on body.site
# plot sp richness based on body site. color points based on the metadata category "subject". 
```{r}
plot_richness(physeq, x="body.site", color="subject", 
measures=c("Observed", "Simpson", "Shannon"))
```

# plot alpha diversity based on reported.antibiotic.usage
# could do x="subject"
```{r}
plot_richness(physeq, x="reported.antibiotic.usage", color="body.site", measures=c("Observed", "Simpson", "Shannon"))
```

# plot alpha diversity based on subject
```{r}
plot_richness(physeq, x="subject", shape="body.site", 
measures=c("Observed", "Simpson", "Shannon"))
```


# test for normality
# In sci you usually have null and alternative hypoth. the null is that the data is normally distributed. the alt is that the data is not normally distributed. if p val is less than 0.05, we can say that the data is not normally distributed.
```{r}
#create an object called alpha. the function estimate_richness : for every sample, it pulls out the richness categories we were interested in. Observed spp, and it'll calc the simpson and shannon for every sample. 
#four column. one column will be observed, shannon, and simpson number.
# creates 34 objects with 3 variables
alpha <- estimate_richness(physeq, measures=c("Observed", "Simpson", "Shannon"))

# Shapiro-wilk. a test for normality when you have less than 50 samples. 
# you do shapiro test on each one of the columns. store results in three diff objects. next line says please do shapiro.test on alpha object, ($=column) on the column called observed. results from shapiro test
# shannon is normally distribued here. other two are significantly NOT normal.
observed <- shapiro.test(alpha$Observed)
shannon <- shapiro.test(alpha$Shannon)
simpson <- shapiro.test(alpha$Simpson)
```

#create data frames for statistacal analyses. perform diff. statistical analyses
# merge metadata file with diversity indexes (observed, shannon, simpson)
# changes the alpha file!
# here, we are using raw data
```{r}
#pull out all sample info/sample names from phyloseq object. 
#extract sample information from the physeq object

samples <- sample_data(physeq)

# If samples is a phyloseq sample_data object, convert it to a data frame
if (class(samples) =="sample_data") {
  samples <- data.frame(sample_data(samples))
}

#convert to data frame. a table. a format R can work with.
#add a column to alpha with sample names
alpha$sample <- rownames(alpha)

#merge sample name/info file with alpha file we generated
#merge alpha diversity and sample data.
alpha <- merge(alpha, samples, by = "sample")
```

# perform statistics based on subject
# output values greater than p=0.05
```{r}
# Perform t/wilcox tests for each biodiversity index
# create three objects. 
# do t test for shannon diversirt based on subject because data is normally distributed
test_observed <- wilcox.test(Observed ~ subject, data = alpha)
test_simpson <- wilcox.test(Simpson ~ subject, data = alpha)
test_shannon <- t.test(Shannon ~ subject, data = alpha)

# Printing the results
print(test_observed)
print(test_simpson)
print(test_shannon)
```


# perform statistics based on reported.antibiotic.usage
# output values less than p=0.05
```{r}
# Perform t/wilcox tests for each biodiversity index
# create three objects. 
# do t test for shannon diversirt based on reported.antibiotic.usage because data is normally distributed
test_observed <- wilcox.test(Observed ~ reported.antibiotic.usage, data = alpha)
test_simpson <- wilcox.test(Simpson ~ reported.antibiotic.usage, data = alpha)
test_shannon <- t.test(Shannon ~ reported.antibiotic.usage, data = alpha)

# Printing the results
print(test_observed)
print(test_simpson)
print(test_shannon)
```

#test for body site
# Anova for normally distributed data. If not, do kruskal-wallace test
# this code is doing a multiple comparison test. compairing 4 diff body sites. just doing an anova or just krusk-wall will only tell us that there is a sig. diff. But we want to know how. in what ways they're diff. we want to kno if its the gut and palm, gut mouth, etc. 
# we are doing a post-hoc test. for non-normally distributed data, we do a pairwise. When doing anova, we do a tookie test.
```{r}
kruskal.test(Simpson ~ body.site, data=alpha)
pairwise.wilcox.test(alpha$Simpson, alpha$body.site, p.adjust.method="holm")
kruskal.test(Observed ~ body.site, data=alpha)
pairwise.wilcox.test(alpha$Observed, alpha$body.site, p.adjust.method="holm")
shannonanova <- aov(Shannon ~ body.site, data=alpha)
summary(shannonanova)
TukeyHSD(shannonanova)
```